File: back/models/attend.go
package models

import (
	"time"
)

type Attend struct {
	UserID      int       `gorm:"primaryKey" json:"user_id"`
	User        *User     `gorm:"foreignKey:UserID" json:"user,omitempty"`
	EventID     int       `gorm:"primaryKey" json:"event_id"`
	Event       *Event    `gorm:"foreignKey:EventID" json:"event,omitempty"`
	HasAttended bool      `gorm:"default:false;not null" json:"has_attended"`
	CreatedAt   time.Time `json:"created_at"`
	UpdatedAt   time.Time `json:"updated_at"`
}

File: front/lib/core/models/attend.dart
import "package:front/core/models/user.dart";

import "event.dart";

class Attend {
  final int userId;
  final User? user;
  final int eventId;
  final Event? event;
  final bool hasAttended;
  final DateTime createdAt;
  final DateTime updatedAt;

  Attend({
    required this.userId,
    this.user,
    required this.eventId,
    this.event,
    required this.hasAttended,
    required this.createdAt,
    required this.updatedAt,
  });

  factory Attend.fromJson(Map<String, dynamic> json) {
    return Attend(
      userId: json['user_id'],
      user: json['user'] != null ? User.fromJson(json['user']) : null,
      eventId: json['event_id'],
      event: json['event'] != null ? Event.fromJson(json['event']) : null,
      hasAttended: json['has_attended'],
      createdAt: DateTime.parse(json['created_at']),
      updatedAt: DateTime.parse(json['updated_at']),
    );
  }
}
